### DISCUSSION TITLE
Como listar os parâmetros de minha API

### DISCUSSION THREAD

**Discussion by AlexhaifaGarcia** [HashLoad/horse#388] [General]
Boa Tarde,

Analisando o Req.Query do horse, eu tenho a opção de Req.Query.Count que me informa a quantidade de parâmetros que chegou na minha API, porém não encontrei uma forma onde eu poderia fazer um for para pegar os nomes dos parâmetros por index, tipo:

var
  i : integer;

Begin 
  for i := 0 to Pred( Req.Query.Count ) do 
  Begin 
     
      Memo1.line.add( Req.Query.Item[i] ) // algo assim 
      

  End;
End;

Poderia me dar alguma dica?
Obrigado

**Comment by gabrielbaltazar** [comment]
Boa tarde.

Você poderia fazer algo assim por exemplo

```delphi
var
  LNome: string;
  LValor: string;
begin
  for LNome in Req.Query.Dictionary.Keys do
  begin
    LValor := Request.Query.Field(LNome).AsString;
    //
  end;
```

**Reply by AlexhaifaGarcia** [reply]
Boa Tarde Gabriel.

Obrigado pela dica, irei fazer esse teste assim que puder.

Muito obrigado pela dica.

**Comment by AlexhaifaGarcia** [comment]
Olá Gabriel, deu certo.
Fiz o processo pelo dicionário mesmo. 
Muito obrigado pela dica
